lib/index.js

```diff
@@ -307,8 +314,9 @@ export function configureUnifiedLanguageServer(
     const files = await processDocuments(textDocuments)
 
     for (const file of files) {
-      // VFile uses a file path, but LSP expects a file URL as a string.
-      const uri = String(pathToFileURL(file.path))
+      // VFile.path for URI restoration is not suitable here because it
+      // resolves symlinks
+      const uri = String(file.data.uri)
```

Comment:
I have also been fiddling with this, because this turns out to be very closely related to #29.

I think this specific issue can be resolved using the following change:

```suggestion
      // unified-engine returns a relative path, but LSP expects an absolute file URL as a string.
      const uri = String(pathToFileURL(path.resolve(file.cwd, file.path)))
```

This means the custom data can be removed.

Reply:
I changed my mind about this. Due to various internal transforms (url ↔ path, `/` ↔ path separator, case insensitive file systems), I now think it makes perfect sense to keep the original document uri around to use for reporting back.